src/query/interpreter.cpp

```diff
@@ -3116,13 +3117,16 @@ PreparedQuery PrepareSystemInfoQuery(ParsedQuery parsed_query, bool in_explicit_
       handler = [storage = current_db.db_acc_->get()->storage(), interpreter_isolation_level,
                  next_transaction_isolation_level] {
         auto info = storage->GetBaseInfo();
+        const auto vm_max_map_count = utils::GetVmMaxMapCount();
```

Comment:
Maybe it makes sense just to pass back an integer instead of the optional<int>, since everywhere where we use it we check if it has a value and set it to `-1` in case there is no value.

Reply:
I would leave it like this because in memgraph.cpp I am getting value, and I have different logic for communicating to user that I couldn't read `vm_max_map_count`. If I return -1, then instead of checking if it has value, I need to check for -1, which is special number in this case, and could be error prone, if someone later decides to return 0 as special number.